user settings: Fixing download zuliprc button.

dipu989 opened the pull request:
<!-- What's this PR for?  (Just a link to an issue is fine.) -->
Fixes:- https://github.com/zulip/zulip/issues/10597

**Testing Plan:** <!-- How have you tested? -->
Tested in the local development environment.
All Ok!

**GIFs or Screenshots:** <!-- If a UI change.  See:
  https://zulip.readthedocs.io/en/latest/tutorials/screenshot-and-gif-software.html
  -->
Before Changes:-
![bug](https://user-images.githubusercontent.com/31280303/46262352-976c9800-c51d-11e8-9836-cb58e4c72cd9.png)

After Changes:-
1) Maximum and medium screen sizes:-
![maxmid](https://user-images.githubusercontent.com/31280303/46262367-bd923800-c51d-11e8-82b1-196952b39763.png)

2) Smaller screen size:-
![mini](https://user-images.githubusercontent.com/31280303/46262380-e3b7d800-c51d-11e8-8a4e-39e112cb690b.png)


<!-- Also be sure to make clear, coherent commits:
  https://zulip.readthedocs.io/en/latest/contributing/version-control.html
  -->

dipu989 commented:
@synicalsyntax 
The Bug was pointed out by @akashnimare.
Ready for review and open to suggestions.

timabbott commented on static/styles/settings.scss:
```diff
@@ -1008,6 +1008,10 @@ input[type=checkbox].inline-block {
 
+.flexbox div {
+    display: inline-flex;
+}
```
Either this belongs in `components.scss` (after a lot of careful thought about potential implications) or it should be scoped to the component it's intended to modify.  Otherwise, it'll potentially break other parts of the site.  

@synicalsyntax can you help @dipu989 get this PR in order?

synicalsyntax replied:
Agreeing with Tim that `.flexbox` isn't being used properly here; is there a reason why you are creating a div element for each item in the flexbox? They can be aligned in better ways.

synicalsyntax replied:
1. This should probably be `flex`, not `inline-flex` (they're block vs inline-block flexbox equivalents)
2. Can you find a more descriptive class name than `flexbox`? It's pretty vague for a specific element, and if we were to turn flexbox into a component class in the future, this would ruin it. A better identifier might be an ID named api_key_buttons.

dipu989 replied:
@synicalsyntax 

1) Changing inline-flex to flex is causing elements to take up the entire space present in the area. That's why I have left inline-flex as it is(in the new commit) . But if it's causing problem, then i would change display to flex and at smaller screen size, display will be set to inline flex. I have checked this approach and it's working fine.

2) Changed the class flexbox to id api_key_buttons.

dipu989 replied:
@synicalsyntax  @timabbott 
PR is ready for review.

dipu989 commented:
@timabbott  @synicalsyntax 
I have updated the PR with some changes.
They are as:-
1) I have selected the button using class (.regenerate_api_key), so it only affects that button and not any other parts of the site.
2) Removed unnecessary div elements.

synicalsyntax commented on static/templates/settings/account-settings.handlebars:
```diff
@@ -212,13 +212,15 @@
+                <a id="download_zuliprc" download="{{zuliprc}}" class="button sea-green">
+                    {{t "Download .zuliprc" }}
+                </a>
+                <div id="user_api_key_error text-error">
```
I don't think this is valid HTML (you can't have two IDs for an element). Shouldn't text-error be a class?

synicalsyntax replied:
Once you resolve the `text-error` issue, you'll find that the error is being incorrectly displayed inline with the buttons (it should be on a separate line under). Can you correct this regression?

![screenshot at oct 13 16-40-58](https://user-images.githubusercontent.com/15116870/46911102-f7dcdb80-cf06-11e8-94cf-a2e880dbaa61.png)

dipu989 replied:
@synicalsyntax 

1) Changed text-error as class.

2)  text-error issue resolved.

![error](https://user-images.githubusercontent.com/31280303/46917185-802faf00-cfe1-11e8-8908-85bc83dd76d0.png)

timabbott commented:
@dipu989 it looks like you didn't post a comment after fixing these issues, so I didn't end up following up with another review.  Sorry about that!  In general, on GitHub, you should always post when you're ready for another review.
